portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,81 @@
 
 package com.google.sps.servlets;
 
+import java.util.*; 
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns "Hello Jay Saleh!". TODO: modify this file to handle comments data */
+/** Servlet that writes an list of messages as a response. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
+  /** Creates a "comment" entity and stores in database */
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = getParameter(request, /* name= */ "name-input", /* defaultValue= */ "");
+    String commentText = getParameter(request, /* name= */ "comment-input", /* defaultValue= */ "");
+    long timeStamp = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("name", name);
+    commentEntity.setProperty("commentText", commentText);
+    commentEntity.setProperty("timeStamp", timeStamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    response.sendRedirect("/html/comments.html");
+  }
+
+  /**
+   * Gets value from form in comments section of portfolio and returns it. If form is empty,
```

Comment:
I would rephrase this to be a bit more specific.

Example:
Returns value with {@code name} from the {@code request} form. If the {@code name} cannot be found, then returns {@code defaultValue}.

By tagging with @code you do not need to mention defaultValue was passed in.
Also, nothing in this code is guaranteeing that the request is of the comments section, so that can be omitted.

Reply:
Done.